databus/src/main/java/com/bazaarvoice/emodb/databus/core/DefaultDatabus.java

```diff
@@ -120,6 +120,11 @@
     /* This is how long we submit tasks to drain the queue for each subscription from one poll request */
     private static final Duration MAX_QUEUE_DRAIN_TIME_FOR_A_SUBSCRIPTION = Duration.ofMinutes(1);
 
+    /* We don't allow subscriptions to be created beyond this number. */
+    private static final Duration MAX_SUBSCRIPTION_TTL = Duration.ofDays(365 * 10);
```

Comment:
I know you made this 10 years to protect the replay subscription, but this kind of negates the purpose of the PR. With this check, somebody can still create a subscription that can last up to **10 years**

Reply:
still, PR started with the purpose of protecting Event TTL.